__tests__/server/middleware/pwa/service-worker.spec.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  serviceWorkerMiddleware,
+  getServiceWorkerEnabled,
+  getServiceWorkerScript,
+  getServiceWorkerScope,
+  getServiceWorkerType,
+  setServiceWorkerScript,
+  configureServiceWorker,
+  createServiceWorkerScript,
+  createServiceWorkerEscapeHatchScript,
+  createServiceWorkerNoopScript,
+} from '../../../../src/server/middleware/pwa/service-worker';
+
+jest.mock('fs', () => ({
+  readFileSync: (filePath) => ({ toString: () => (filePath.endsWith('noop.js') ? '[service-worker-noop-script]' : '[service-worker-script]') }),
+}));
+
+describe('service worker middleware components', () => {
+  test('exports functions as default', () => {
+    expect.assertions(10);
+    expect(serviceWorkerMiddleware).toBeInstanceOf(Function);
+    expect(getServiceWorkerEnabled).toBeInstanceOf(Function);
+    expect(getServiceWorkerScope).toBeInstanceOf(Function);
+    expect(getServiceWorkerType).toBeInstanceOf(Function);
+    expect(getServiceWorkerScript).toBeInstanceOf(Function);
+    expect(setServiceWorkerScript).toBeInstanceOf(Function);
+    expect(configureServiceWorker).toBeInstanceOf(Function);
+    expect(createServiceWorkerScript).toBeInstanceOf(Function);
+    expect(createServiceWorkerEscapeHatchScript).toBeInstanceOf(Function);
+    expect(createServiceWorkerNoopScript).toBeInstanceOf(Function);
+  });
+
+  test('gets default values', () => {
+    expect.assertions(4);
+    expect(getServiceWorkerEnabled()).toBe(false);
+    expect(getServiceWorkerScript()).toBe(null);
+    expect(getServiceWorkerScope()).toBe(null);
+    expect(getServiceWorkerType()).toBe(null);
+  });
+
+  test('sets the service worker script and scope', () => {
+    expect.assertions(3);
```

Comment:
not harmful at all but `expect.assertions` is typically only needed in async tests

Reply:
I've removed a few of them as they were unnecessary